### PR TITLE
Clippy lint

### PR DESCRIPTION
**Pull request by Mathieu-Lala**
* turn on clippy all / nursery / pedantic / cargo
* fix all warning
* allow when fix would make too much changes
* bumps dependencies version
* no meaningfull changes in the program logics
* remove unused files from previous version

### CODE REVIEW

File: src/rules/rule_engine.rs [hunk 1]
```diff
@@ -189,8 +190,8 @@ impl RuleEngine {
             Err(error) => {
                 log::error!(
                     target: SRULES,
-                    "stage '{}' skipped => rule engine failed to evaluate rules:\n\t{}",
-                    stage,
+                    "smtp_stage '{}' skipped => rule engine failed to evaluate rules:\n\t{}",
```
Comment (ltabis) [inline src/rules/rule_engine.rs:193]: `smtp_stage` should be just `stage`, or `smtp stage`

File: src/smtp/code.rs [hunk 2]
```diff
@@ -132,7 +132,7 @@ pub enum SMTPReplyCode {
 }
 
 impl SMTPReplyCode {
-    pub(crate) fn is_error(&self) -> bool {
+    pub(crate) fn is_error(self) -> bool {
```
Comment (ltabis) [inline src/smtp/code.rs:135]: Same as previous comments, the instance would be consumed.

**Review by ltabis — CHANGES_REQUESTED** [review]
Please check out the comments and tell me what you think.

File: src/config/service.rs [hunk 3]
```diff
@@ -4,31 +4,32 @@ use crate::{
...
 }
 
 impl ServiceResult {
-    pub(crate) fn new(status: std::process::ExitStatus) -> Self {
+    pub(crate) const fn new(status: std::process::ExitStatus) -> Self {
         Self { status }
     }
 
-    pub(crate) fn has_code(&self) -> bool {
+    pub(crate) fn has_code(self) -> bool {
```
[the remark this answers is not in the document]
Comment (Mathieu-Lala) [reply src/config/service.rs:19]: The struct derive copy, and the sizeof the struct is small.
So taking a copy is lighter than take a reference

File: src/rules/rule_engine.rs (hunk 1 above)
Comment (Mathieu-Lala) [reply src/rules/rule_engine.rs:193]: See https://rust-lang.github.io/rust-clippy/master/#similar_names

File: src/config/service.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (ltabis) [reply src/config/service.rs:19]: Ok! Do you know what clippy lint changed this ?

File: src/rules/rule_engine.rs (hunk 1 above)
Comment (ltabis) [reply src/rules/rule_engine.rs:193]: Got it, I'll just change the error message then and keep `smtp_stage`.

**Review by ltabis — APPROVED** [review]
All good

**Pull request merged**